lib/source-types.nix

```
@@ -0,0 +1,25 @@
+{ lib }:
+
+lib.mapAttrs (tname: tset: let
+  defaultSourceType = {
+    shortName = tname;
+    isSource = false;
+  };
+
+  mkSourceType = sourceTypeDeclaration: let
+    applyDefaults = sourceType: defaultSourceType // sourceType;
+  in lib.pipe sourceTypeDeclaration [
+    applyDefaults
+  ];
+in mkSourceType tset) {
```

I find that that is a very convoluted way to say. Let's add additional complexity when we need it!

```suggestion
let
  defaultSourceType = tname: {
    shortName = tname;
    isSource = false;
  };
in lib.mapAttrs (tname: tset: defaultSourceType tname // tset) {
```

---

This is a perfect example of where I said

> This is heavily based on patterns used by the meta.licenses infrastructure, so may appear overengineered for its initial level of use. I felt it was more important to follow existing patterns than go for simplicity and risk the two features evolving differently.

I'm trying to avoid divergence between the two.

---

@risicle I see. I am not set on changing this into something simpler. We can also leave it as is, as functionality is equivalent.

_That being said, I personally lean towards the reasoning that @a-m-joseph presented._